platform/admin/core/admin-core-impl/src/test/java/org/codice/ddf/admin/core/impl/SystemPropertiesAdminTest.java

```diff
@@ -208,6 +183,31 @@ public void testWriteSystemProperties() throws Exception {
     }
   }
 
+  @Test
+  public void testWritePlaceholderProperties() throws Exception {
```

Comment:
I would like to see a unit test where an internal port is changed in the installer but the admins wants the corresponding external port to stay at the default. This is the case that failed before your PR because the external ports reference the internal ports.

Reply:
My bad, it's not illustrated well that with this test but this is meant to be that case. 
The reason it looks so strange is whenever we call write we've already done the read and have all of the default properties. 
Maybe changing it to do the read might be more clear.
Pseudo-code version of this test might be like:
```
Map updatedProps = readSystemProperties(); //get default values
updatedProps.put(newInternalHttpPort); // simulates changing the port in the installer 
```

Reply:
For this case I'm describing where the admin updates just the internal ports in the installer, I would expect `spa` to contain

```java
map.put(SystemBaseUrl.INTERNAL_HTTP_PORT, "<new port>");
map.put(SystemBaseUrl.INTERNAL_HTTPS_PORT, "<another new port>");
map.put(SystemBaseUrl.EXTERNAL_HTTP_PORT, "8181"); // the original port in custom.system.properties
map.put(SystemBaseUrl.EXTERNAL_HTTPS_PORT, "8993"); // the original port in custom.system.properties
```

I don't think we need to mess with readSystemProperties. `custom.system.properties` for a default DDF install contains
```
org.codice.ddf.external.httpsPort=${org.codice.ddf.system.httpsPort}
org.codice.ddf.external.httpPort=${org.codice.ddf.system.httpPort}
```
I want to confirm that `org.codice.ddf.external.httpsPort` and `org.codice.ddf.external.httpPort` are loaded as their original values (8993 and 8181) after restarted even if the internal ports are changed in the installer. This is a test that would have failed before your PR.

Reply:
Okay this makes sense to me 👍 
I've updated this assertion to be slightly more accurate too : 
```
org.apache.felix.utils.properties.Properties systemProps = new org.apache.felix.utils.properties.Properties(systemPropsFile);
assertThat(systemProps.getRaw(SystemBaseUrl.EXTERNAL_HOST).get(0), equalTo("org.codice.ddf.external.hostname = localhost"));
```
where getRaw gets what's _actually_ in the file.